File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/domain/interactor/uploadImage/UploadImageUseCase.java
package com.lattis.ellipse.domain.interactor.uploadImage;

import com.lattis.ellipse.data.network.model.response.uploadImage.UploadImageResponse;
import com.lattis.ellipse.domain.executor.PostExecutionThread;
import com.lattis.ellipse.domain.executor.ThreadExecutor;
import com.lattis.ellipse.domain.interactor.UseCase;
import com.lattis.ellipse.domain.repository.UploadImageRepository;

import javax.inject.Inject;

import okhttp3.MultipartBody;
import okhttp3.RequestBody;
import io.reactivex.Observable;

/**
 * Created by Velo Labs Android on 06-04-2017.
 */

public class UploadImageUseCase extends UseCase<UploadImageResponse> {
    UploadImageRepository uploadImageRepository;
    private String uploadType;
    private MultipartBody.Part file;
    RequestBody   fileName;

    @Inject
    protected UploadImageUseCase(ThreadExecutor threadExecutor, PostExecutionThread postExecutionThread,UploadImageRepository uploadImageRepository) {
        super(threadExecutor, postExecutionThread);
        this.uploadImageRepository = uploadImageRepository;
    }

    public UploadImageUseCase withUploadType(String uploadType) {
        this.uploadType = uploadType;
        return this;
    }

    public UploadImageUseCase withFile( MultipartBody.Part file) {
        this.file = file;
        return this;
    }

    @Override
    protected Observable<UploadImageResponse> buildUseCaseObservable() {
        return uploadImageRepository.uploadImage(uploadType, file);
    }


}

File: velolabs_repos/lattis_android_v2/domain/src/main/java/com/lattis/domain/usecase/uploadimage/UploadImageUseCase.kt
package com.lattis.domain.usecase.uploadimage

import com.lattis.domain.executor.PostExecutionThread
import com.lattis.domain.executor.ThreadExecutor
import com.lattis.domain.models.UploadImage
import com.lattis.domain.repository.UploadImageRepository
import com.lattis.domain.usecase.base.UseCase
import io.reactivex.rxjava3.core.Observable
import javax.inject.Inject

class UploadImageUseCase @Inject protected constructor(
    threadExecutor: ThreadExecutor,
    postExecutionThread: PostExecutionThread,
    var uploadImageRepository: UploadImageRepository
) : UseCase<UploadImage>(threadExecutor, postExecutionThread) {
    private var uploadType: String? = null
    private var filepath: String? = null
    fun withUploadType(uploadType: String): UploadImageUseCase {
        this.uploadType = uploadType
        return this
    }

    fun withFilePath(filepath: String): UploadImageUseCase {
        this.filepath = filepath
        return this
    }

    override fun buildUseCaseObservable(): Observable<UploadImage> {
        return uploadImageRepository.uploadImage(uploadType!!, filepath!!)
    }

}